### ISSUE TITLE
Friends View -> Network

### ISSUE THREAD

**Issue by defnax** [RetroShare/RSNewWebUI#36]
Friends Webui part isnt yet finished.
at the moment it uses the full window on Friends page.

missed things :

- Start a Private Chat  (direct chat)
- Start a Mail (direct mail)
- View Friends details (show infos on right side)
- Show Private Chat Conversation/History on right side

we need on left side the Friendslist, not sure if groups still work with the rs api?
http://www.cetaceanstechlab.com/flingo/tf-demo/dark-default.html
http://preview.themeforest.net/item/quicky-html-chat-template/full_screen_preview/27127279?_ga=2.96927583.1930035690.1617130766-1181836442.1617130766

mokup idea:
![image](https://user-images.githubusercontent.com/9952056/116786893-6cfe6500-aaa1-11eb-865d-f08b25b4800f.png)

**Comment by csoler** [comment]
Direct chat and direct mail are *deprecated*. So Chat and mail should be only visible in the people tab.

**Comment by csoler** [comment]
What you call "friend" tab is the network tab, where you see "nodes" not "friends".

**Comment by defnax** [comment]
At the moment its enabled on windows and mac builds 
When enabled then would be cool we can chat or mail from webui too,
When anytime the webui will work without qt ui, would be very usefull we have it there too

**Comment by zelfroster** [comment]
The mock up is a great Help, But I was actually talking about this Network Section
![image](https://user-images.githubusercontent.com/93514460/224403739-bb39f487-1de6-43f0-a1d3-c97393e57e2f.png)

**Comment by defnax** [comment]
> The mock up is a great Help, But I was actually talking about this Network Section ![image](https://user-images.githubusercontent.com/93514460/224403739-bb39f487-1de6-43f0-a1d3-c97393e57e2f.png)

yes the network needs a better design like on social networks mostly you view the groups, chat or contacts list on the left side, middle is often chat history or view the details.
similiar like the qt ui, maybe i can be done better

**Comment by defnax** [comment]
![image](https://user-images.githubusercontent.com/9952056/224405245-7333299d-01b3-4a91-ae41-38fd54d28330.png)

we use retroshare for discussion
you we exchange rs ids then we can stay better with other devs in rs dev network

**Comment by defnax** [comment]
send me your full rs key my dht not more works maybe it will not work
ABB/5a5fZwNOGJk25a9Md4tHAxRFglDq7yKGzto/oSH3Nlmz2M0towEGZGVmbmF4kwakAKjAkseSBrrnBS6SxwQD2bvK

**Comment by zelfroster** [comment]
> 

How to get that? 😅

**Comment by defnax** [comment]
![image](https://user-images.githubusercontent.com/9952056/224407367-d54cf40e-1d98-4afb-92b5-a223fed458bb.png)

**Comment by zelfroster** [comment]
Got it 👍🏼 


CQEGAcGbxsBNBGQHRcEBCADFXELmVgPastpwc6DP5MI9NKCdLq2g0aJtVsyDE9Mc
DDTYUoG9//+0WIavmhmBYGqx0ry8XyMy1Epqa7LDBvNK++82A/XSg1QXW5aT+usU
YpfXAS+CRul0DVB7K30FnAuFN9H1V/Be3BE+v4H+LkXuBgE4/ZVbHH5Rf4N1QSkE
4svZxK3ymHMNH+W7cm5k6WrVTF3Eu08R2MvKc9FgVkr4+DKLfaIkkHFlBraoj8y5
XP5cpZz4tSlPLAV4BFZJ7HjmdLzYHqCX54CuGCOfFJZlXmSs/siZ3L4roJtxp+ko
qzaOdZWZpFdAa68i/FPJxC53ExODvRshkVSeg1qHfMU7ABEBAAHNJ3plbGZyb3N0
ZXIgKEdlbmVyYXRlZCBieSBSZXRyb1NoYXJlKSA8PsLAXwQTAQIAEwUCZAdFwQkQ
AxOEg5K0aXYCGQEAALl/B/0StaAZCWW+OXk+yyaiHGQ6huyGemGK2XB933bCVA1h
OMJJf4/hEsTPu6jGpR30AvNMtIGyBurMXH6GFqGP35ePr2JePPIk2AHXTeh61FzV
nCsf6IjiIW1zs5cPIN1EQWDBh90HPEZ6896Xx+EodqZnAu2L94eIMBOIYiDsvrIE
4K2WRurlDu1PXt8uByh+nPbGKCa/GW8pl4TymRKwsIjhrVwkA6NdtEGE4TrXETRR
Qxol0I+mWJWHRsYo/j114MDtvg5WXSYFZZFymvgyqVWvMVP8oqF7Mdplb30dbHgF
8eBajzXsd4q2M9A7pwDvPJxqYbz0FC+ovXuXOue6uYvVAgYAAAAAAAADBsCoAQbp
4QYLTXkgY29tcHV0ZXIFEDl4wBMBBr558c03wzr+z6UHAwMpLg==

**Comment by defnax** [comment]
delete your posts with your keys here, i see no ips in your key, you added me?

**Comment by zelfroster** [comment]
Yes I did. But maybe It didn't get added as my RS got closed unexpectedly.

**Comment by defnax** [comment]
but your key does not contain a external ip and port maybe you need enable include with ip on, when dht not works each other needs the ip/port to know in the
![image](https://user-images.githubusercontent.com/9952056/224413694-fd3b7c0f-be1e-4774-8a77-7bf30fcf70b5.png)

**Comment by zelfroster** [comment]
Ohh Now I did it.


CQEGAcGbxsBNBGQHRcEBCADFXELmVgPastpwc6DP5MI9NKCdLq2g0aJtVsyDE9Mc
DDTYUoG9//+0WIavmhmBYGqx0ry8XyMy1Epqa7LDBvNK++82A/XSg1QXW5aT+usU
YpfXAS+CRul0DVB7K30FnAuFN9H1V/Be3BE+v4H+LkXuBgE4/ZVbHH5Rf4N1QSkE
4svZxK3ymHMNH+W7cm5k6WrVTF3Eu08R2MvKc9FgVkr4+DKLfaIkkHFlBraoj8y5
XP5cpZz4tSlPLAV4BFZJ7HjmdLzYHqCX54CuGCOfFJZlXmSs/siZ3L4roJtxp+ko
qzaOdZWZpFdAa68i/FPJxC53ExODvRshkVSeg1qHfMU7ABEBAAHNJ3plbGZyb3N0
ZXIgKEdlbmVyYXRlZCBieSBSZXRyb1NoYXJlKSA8PsLAXwQTAQIAEwUCZAdFwQkQ
AxOEg5K0aXYCGQEAALl/B/0StaAZCWW+OXk+yyaiHGQ6huyGemGK2XB933bCVA1h
OMJJf4/hEsTPu6jGpR30AvNMtIGyBurMXH6GFqGP35ePr2JePPIk2AHXTeh61FzV
nCsf6IjiIW1zs5cPIN1EQWDBh90HPEZ6896Xx+EodqZnAu2L94eIMBOIYiDsvrIE
4K2WRurlDu1PXt8uByh+nPbGKCa/GW8pl4TymRKwsIjhrVwkA6NdtEGE4TrXETRR
Qxol0I+mWJWHRsYo/j114MDtvg5WXSYFZZFymvgyqVWvMVP8oqF7Mdplb30dbHgF
8eBajzXsd4q2M9A7pwDvPJxqYbz0FC+ovXuXOue6uYvVAgYAAAAAAAADBsCoAQbp
4QYLTXkgY29tcHV0ZXIFEDl4wBMBBr558c03wzr+z6UKGGlwdjQ6Ly8xOTIuMTY4
LjEuNjo1OTg3MwcDekBn

**Comment by zelfroster** [comment]
Also, Please send your key again. I think I need to add you again 😅

**Comment by defnax** [comment]
CQEGAcGXxsBNBFMy+PoBCAC5nsux4tYnNGOrrp2SJ78NyJWST7rUkTs3F6NdB7Kr
sM+rj9x8P2xhtwobLmMxsP6Cag6Rxn0kbzfUkudm7mL4nsVcflR+gdOH8x7GYsnh
V2jZZ1ICmVNYptorcNnSRh2VOI5ACqY+0RMuv7Httq+/LYWhYwEo3hJwMe8VLjS0
UMmQYJKdqLcWfGViGOhhsoId5ezlnPbk/20WIr/UDnqcWcGkoxnGE2ZKc+Mof0RU
7k9HB/+GOBKV6ayHZOk/ysLiuSiN2rHZz+rj2i4thrOhHvtLBPL8iyO8CSw37YE1
+23H2yciLWkIgokZa2OAw+lRS0K2EfRMP49oxNpcdKEbABEBAAHNI2RlZm5heCAo
R2VuZXJhdGVkIGJ5IFJldHJvU2hhcmUpIDw+wsBfBBMBAgATBQJTMvj6CRD3Nlmz
2M0towIZAQAAJMcH/jR1ztoHE8IbORdAVeeaaqf5btKofyR9nOJerGjZ7C76UkqH
RNI+taclUyVp6AmedP9TKyQcOwYv6TV9+lAhXlm+fGw8vrH7Lwrwmqf2nPjKlbXa
oyk361TkWIqwr0oJNSK74ILWNgVuoM9k2b++mVgC6ay3gvDt+2jJATIbY5E8fQQN
o8F3AI3uuvu1mslWQ1gk7NeSDxxiCpvqG8UNOmXduMsc41FhoP52qXhFmbFHXlbp
b0G6e4FOTpK8NHKuk6xGkBI+IR7bkrGcBb7wMcBp34Crc1NaSG4TlZwZZmNMl5lo
KztN2RWDGsaXiL5XITSsbPqTxwkgxu8U4j6NIswCBi4F57qSxwMGwKgApJLHBAAG
CUN1cmlvc2l0eQUQf+WuX2cDThiZNuWvTHeLRwoZaXB2NDovLzE2OS4yNTQuMTMu
NjozNzU3NQocaXB2NDovLzE2OS4yNTQuMTMwLjEwNjozNzU3NQocaXB2NDovLzE2
OS4yNTQuMTk0LjIzODozNzU3NQoaaXB2NDovLzE2OS4yNTQuNS4xMjc6Mzc1NzUK
GmlwdjQ6Ly8xNjkuMjU0LjguMTgxOjM3NTc1ChppcHY0Oi8vMTkyLjE2OC4wLjE2
NDozNzU3NQoZaXB2NDovLzQ2LjIyMy4xLjE0NDozNzU3NQoZaXB2NDovLzQ2LjUu
MjMwLjE1NTozNzU3NQoZaXB2NDovLzQ2LjUuMjMwLjE2ODozNzU3NQoZaXB2NDov
LzQ2LjUuMjMwLjIyOTozNzU3NQoYaXB2NDovLzQ2LjUuMjMwLjUxOjM3NTc1Chlp
cHY0Oi8vNDYuNS4yMzEuMTA1OjM3NTc1ChhpcHY0Oi8vNDYuNS4yMzEuMTI6Mzc1
NzUKGWlwdjQ6Ly80Ni41LjIzMS4xMzc6Mzc1NzUKGGlwdjQ6Ly80Ni41LjIzMS4x
MzozNzU3NQoZaXB2NDovLzQ2LjUuMjMxLjE4NjozNzU3NQo2aXB2NjovL1syYTAy
OjgwNzA6YTE4NDozMzYwOjI3YzY6MTA5OTpkOGY4OmIzYzVdOjM3NTc1CjZpcHY2
Oi8vWzJhMDI6ODA3MDphMTg0OjMzNjA6NmVkMTpmNDBjOmE3M2M6YjlmNl06Mzc1
NzUKNWlwdjY6Ly9bMmEwMjo4MDcwOmExODQ6MzM2MDo3MDhiOmFlNjU6MzY2OjQz
MzRdOjM3NTc1CjZpcHY2Oi8vWzJhMDI6ODA3MDphMTg0OjMzNjA6YTVmYjpmZmVi
OmU0ZjU6YWNiOV06Mzc1NzUHA7yzVw==

**Comment by defnax** [comment]
maybe check if your network shows your ip address and both ports  are same, on your last rs key it does not contain a ip/port:
![image](https://user-images.githubusercontent.com/9952056/224415358-a1a213bc-5d33-4edf-90de-afd5520475bc.png)

**Comment by defnax** [comment]
i think test to use the external ip finder, that was the reason your rs key dos not contain your external ip and port

**Comment by zelfroster** [comment]
Its working, I got in touch with Csoler via the dht key.

Also, you show up in my people tab.
![1678478838_Mar11_01:37:18](https://user-images.githubusercontent.com/93514460/224418251-6b671466-3705-4048-af22-942e94b802a1.png)

**Comment by zelfroster** [comment]
Also, 

> i think test to use the external ip finder, that was the reason your rs key dos not contain your external ip and port ![image](https://user-images.githubusercontent.com/9952056/224417844-dd9b6bde-fb2a-4e7b-8d28-d4d20cf0dbab.png)
> 
> is your dht working? i think not right

This is already checked.

**Comment by zelfroster** [comment]
Yeah, so I think DHT is not working. 
![1678479256_Mar11_01:44:16](https://user-images.githubusercontent.com/93514460/224419043-1b4dd24f-3e39-4229-bd2a-dfafe3ea8722.png)

**Comment by defnax** [comment]
for me it looks like your dht not working + your rs has not get yet your own external ip &port , when field is empty in network, rs still not connects

**Comment by defnax** [comment]
wait normal the external ip will find it but it can take time

**Comment by zelfroster** [comment]
I used this command to build RS, maybe that's why its not working.

![1678479456_Mar11_01:47:36](https://user-images.githubusercontent.com/93514460/224420320-7777b162-7dce-4a57-aaaa-5188c8d8060a.png)

**Comment by defnax** [comment]
maybe test this option you need to wait till it gets the ip from ip finder
![image](https://user-images.githubusercontent.com/9952056/224420810-c497c67b-d8cb-4c72-804a-1aab355a982a.png)

**Comment by defnax** [comment]
check your email @zelfroster

**Comment by defnax** [comment]
improved here: #119
